src/Appwrite/Event/Realtime.php

```diff
@@ -0,0 +1,172 @@
+    {
+        $this->project = $project;
+        $this->event = $event;
+        $this->payload = new Document($payload);
+    }
+
+    /**
+     * @param string $project
+     * return $this
+     */
+    public function setProject(string $project): self
+    {
+        $this->project = $project;
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getProject(): string
+    {
+        return $this->project;
+    }
+
+    /**
+     * @param string $event
+     * return $this
+     */
+    public function setEvent(string $event): self
+    {
+        $this->event = $event;
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getEvent(): string
+    {
+        return $this->event;
+    }
+
+    /**
+     * @param array $payload
+     * return $this
+     */
+    public function setPayload(array $payload): self
+    {
+        $this->payload = new Document($payload);
+        return $this;
+    }
+
+    /**
+     * @return Document
+     */
+    public function getPayload(): Document
+    {
+        return $this->payload;
+    }
+
+    /**
+     * Populate channels array based on the event name and payload.
+     *
+     * @return void
+     */
+    private function prepareChannels(): void
+    {
+        switch (true) {
+            case strpos($this->event, 'account.') === 0:
+                $this->channels[] = 'account.' . $this->payload->getId();
+                $this->permissions = ['user:' . $this->payload->getId()];
+
+                break;
+            case strpos($this->event, 'database.collections.') === 0:
+                $this->channels[] = 'collections';
+                $this->channels[] = 'collections.' . $this->payload->getId();
+                $this->permissions = $this->payload->getAttribute('$permissions.read');
+
+                break;
+            case strpos($this->event, 'database.documents.') === 0:
+                $this->channels[] = 'documents';
+                $this->channels[] = 'collections.' . $this->payload->getAttribute('$collection') . '.documents';
+                $this->channels[] = 'documents.' . $this->payload->getId();
+                $this->permissions = $this->payload->getAttribute('$permissions.read');
+
+                break;
+            case strpos($this->event, 'storage.') === 0:
+                $this->channels[] = 'files';
+                $this->channels[] = 'files.' . $this->payload->getId();
+                $this->permissions = $this->payload->getAttribute('$permissions.read');
+
+                break;
+        }
+    }
+
+    /**
+     * Execute Event.
+     * 
+     * @return void
+     */
+    public function trigger(): void
+    {
+        $this->prepareChannels();
+        if (empty($this->channels)) return;
+
+        $redis = new \Redis();
+        $redis->connect(App::getEnv('_APP_REDIS_HOST', ''), App::getEnv('_APP_REDIS_PORT', ''));
```

Comment:
It would be better to find a way to reuse current Redis connection rather than start a new one.

Reply:
Yeah, I would have to create a Redis Pool outside of the CoRoutine and pass a Redis Instance to the CoRoutine 👍🏻

Reply:
Have created a Redis Pool for now adding its own registry entry for the realtime.php and access it when the adapter is given the true flag here: https://github.com/appwrite/appwrite/blob/a3fb2abf66c95f138294e46d14a70887a0a11f5f/src/Appwrite/Database/Adapter/Redis.php#L34

Why I haven't switched everything in cache to the connection pool is, that there are some places where it breaks - since it cannot be used outside the swoole coroutine 👍🏻 Need to discuss, since this needs refactoring outside of the projects scope.